### ISSUE TITLE
Remove escaping from external compaction entries in metadata table

### ISSUE THREAD

**Issue by keith-turner** [apache/accumulo#5060]
**Is your feature request related to a problem? Please describe.**

External compaction entiries in the metadata table are json and include the files that are being compacted. The files in the json are also json and when added to the outer json as values are escaped. 

**Describe the solution you'd like**

Not sure if its possible, but it would be nice to remove the escaping for the file entries if possible.

**Describe alternatives you've considered**

Could do nothing, the escaping is very unpleasant to look at when scanning he metadata table but its not hurting anything.  The escaping may make using tools like jq w/ the data a bit harder also.

**Comment by cshannon** [comment]
I took a look at this and was thinking a bit about it and the only really good way to get rid of the escaping would be to not use the serialized json string of the files and instead use an object representation of StoredTabletFile, but that would break the encapsulation of StoredTabletFile and expose the file/range outside of it so I'm not sure if that's a great idea. A custom serializer would be tricky because escaping quotes are needed so it's valid json.

I'm wondering if this is something that is better solved as custom logic when scanning and logging to print it out in an easier to read format vs storing it differently.

**Comment by dlmarion** [comment]
Does it need to be human readable in the metadata table? If not, then we could just store the list of input files as base64 encoded json. We could also use some other type of encoding (like [CBOR](https://cbor.io/)), but if you want to use jq then base64 decoding is easy on the command line.

**Comment by keith-turner** [comment]
> Does it need to be human readable in the metadata table? 

Human readable would be nice.  I opened this issue when attempting to use that information in the metadata table to track down bugs  and finding it really hard to look at w/ all of the escaping.

**Comment by cshannon** [comment]
> > Does it need to be human readable in the metadata table?
> 
> Human readable would be nice. I opened this issue when attempting to use that information in the metadata table to track down bugs and finding it really hard to look at w/ all of the escaping.

I agree, human readable would be nice when scanning. Otherwise we'd need some sort of tool I think to be able to make sense of the data and decode it.

**Comment by dlmarion** [comment]
#5237 was reverted because the exact metadata entry was not being retained in the external compaction json in the metadata table.
